### PR TITLE
feat: payment builder form update

### PR DESCRIPTION
**Pull request by joanna-pagepro**
https://github.com/JoinColony/colonyCDapp/issues/2004

Create expenditure, without preview mode

### CODE REVIEW

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/utils.tsx
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/utils.tsx:103]: that needs be converted to seconds

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Looks good @joanna-pagepro, thanks for your work on this PR.

A few things that I spotted:

1. Not sure about this one, but advanced payment can be created in a specific domain, so we might need to show `Created In` field at all times. Then with reputation decision method, I presume we could use the same domain for both the motion and the payment. @arrenv could you confirm?

image

2. Typing in big numbers causes some overlaps:

image

3. I have about 91M tokens. The validation checks correctly that the single payout amount doesn't exceed the balance (althought the error with the path looks odd):
image

But it doesn't check that the total amount of all payouts doesn't exceed the balance:
image

4. It crashes when doing the following (on an empty form, with just action type selected):
   1. Set the amount to something exceeding the balance.
   2. Click on Add transaction.
   
https://github.com/JoinColony/colonyCDapp/assets/112586815/488a65f8-fb55-480d-946f-a0bb4ce7c0e7

5. The heading of the table should be `Payments` instead of `Additional Payments` according to Figma.

But overall, the payments are being created, so nice work! 👍 
image

image

**Comment by AdrianDudko** [comment]
> 1. Not sure about this one, but advanced payment can be created in a specific domain, so we might need to show `Created In` field at all times. Then with reputation decision method, I presume we could use the same domain for both the motion and the payment. @arrenv could you confirm?
> 
> image


I talked about the Created In field with @arrenv. From what I understand, the field should be hidden. But you're right, it will be good to get confirmation from @arrenv again. Nevertheless, this is a rather minor change and if it does not affect the creation of the action itself (the action passes), I think we can merge this PR after fixing the rest of the errors - so as not to block further work on the Adv.Payment process.

I'll make a note to come back to this when @arrenv  is available.
Are you ok with that?

**Comment by jakubcolony** [comment]
Hey @AdrianDudko, yes I agree that this isn't a blocker for merging this PR.

**Comment by joanna-pagepro** [comment]
@jakubcolony please review again. There are no changes with `createdIn` field for now.

**Comment by AdrianDudko** [comment]
> @jakubcolony please review again. There are no changes with `createdIn` field for now.

Agreed with @arrenv to hide `Created In` for `Decision Method: Permission`

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for making changes @joanna-pagepro. 

A couple of things I noticed:

- The validation for amount happens immediately after selecting a recipient, even though the amount field has not been touched. It only seems to happen if you fill in other fields beforehand:

https://github.com/JoinColony/colonyCDapp/assets/112586815/935c4cc3-2315-4d17-8546-28ba1419f137


- If you click on the warning icon next to recipient, the popover gets clipped by the table:
image


- Clicking on "Add token" throws an error:

https://github.com/JoinColony/colonyCDapp/assets/112586815/5bf7f0c8-051a-4291-808b-305270b2d018

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @joanna-pagepro! Nice start on the payment builder!

Some feedback for this one:

1. The enter amount text should not show and should show a 0 as per Figma:
image

https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=3889-462623&mode=design&t=BNXdv8DRozb21TVR-4
 
 2. When I select a different token that has no funds, the validation shows displaying a message that the amount should be greater than 0 but there are no funds.
 
image
 
 Then when I select back to the token that has funds available, the validation stays and does not hide. This is a confusing user experience. 
image

4. The column widths of the table can be match better to Figma which should prevent data from being displayed in a small area
https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=3889-462623&mode=design&t=BNXdv8DRozb21TVR-4 
 
 
 5. When I select 'add token' it breaks
image

6. The claim delay breaks when a section row is added
image

7. The claim delay placeholder is not styled correctly to match Figma
image

9. The styling of the alert and validated user icons int he table are two big and the padding is incorrect 
image

Figma: https://www.figma.com/file/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?type=design&node-id=1782-151741&mode=design&t=FacBc4ZDz7L2AA0Q-4

10. Button should be 'add payment' not 'add transaction' 

11. When I have 1 row with an error validation, as soon I select to add more rows, the error validation continues into those new rows added. The validation should clear for the new rows and stay for the wrong row only
image

12. Number formatting is not consistent int he table with two rows treating numbers differently 
image

13. On mobile, the Hours font changes
image

14. On mobile, there should be no HR's between all individual rows.

**Comment by joanna-pagepro** [comment]
@jakubcolony @melyndav form updated, please review again.
@melyndav I don't know what you mean in point 12. Can you explain more, what do you expect?

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderTokensTotal/types.ts
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderTokensTotal/types.ts:5]: The domain term for those would be "payouts", I think it's worth updating it here to avoid confusion with the `Token` type (which is a property rather than a parent of a payout)

File: src/utils/validation/amountGreaterThanZeroValidation.ts
Comment (jakubcolony) [inline src/utils/validation/amountGreaterThanZeroValidation.ts:30]: There's a util for that in `~utils/tokens.ts`: `getSelectedToken`

File: src/components/v5/common/Table/utils.tsx
Comment (jakubcolony) [inline src/components/v5/common/Table/utils.tsx:37]: Setting different z-indexes based on the screen size makes it look like we've got a bigger problem presumably with another component depending on this? Is it possible to come up with a single value?

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for another round of fixes @joanna-pagepro. All the previous issues seems to be resolved 👍 

I left a few comments and noticed two more things:

- The error message is not human-friendly when claim delay is too set too large:
image

- The totals aren't calculated until you select a domain (which I don't think is necessary to calculate just the total amount of payments):
image

**Comment by joanna-pagepro** [comment]
Some features stopped working after the rebase with the parent branch. Please hold off on Code Review until it's fixed.

**Comment by melyndav** [comment]
@melyndav I don't know what you mean in point 12. Can you explain more, what do you expect?

Hey @joanna-pagepro I've discussed this with Arren and please ignore this point for now since it's a wider decision on number formatting. Thanks!

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderRecipientsField/hooks.tsx
Comment (joanna-pagepro) [reply src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderRecipientsField/hooks.tsx:31]: In this case, we will have to add eslint ignore rule and because of that we won't see if anything is missing from the dependency array.

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderTokensTotal/types.ts
Comment (joanna-pagepro) [reply src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderTokensTotal/types.ts:5]: Do you mean that I should replace `amount` with `payouts` or something else?

File: src/components/v5/common/Table/utils.tsx
Comment (joanna-pagepro) [reply src/components/v5/common/Table/utils.tsx]: It's not needed here anymore. I forgot to remove it.

**Comment by joanna-pagepro** [comment]
@jakubcolony @melyndav it's ready to review again :)

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderRecipientsField/hooks.tsx
Comment (jakubcolony) [reply src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderRecipientsField/hooks.tsx:28]: Sure, I think that's cleaner as you know straight away what's going on

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @joanna-pagepro! Great work on these updates. Some more feedback.

1. The hours input should be styled the same as the amount in the default state.
image

2. members don't show with verified icons in the select dropdown. It shows only in the table. 
image

3. I spoke with @arrenv regarding the claim day max. Right now we have it set to a max of 4 (9999 hrs). We'd like to increase this to 5 (99 999hrs). Could you update this? 

4. In the kebab menu, can you please change the following text 'Add token' to 'Add row'. The functionality remains the same.

**Comment by joanna-pagepro** [comment]
@melyndav all updated, please check if it's ok now

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/ActionSidebar.tsx:121]: If possible, it would be good not to change z-index depending on the screen width

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderPayoutsTotal/PaymentBuilderPayoutsTotal.tsx
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderPayoutsTotal/PaymentBuilderPayoutsTotal.tsx:30]: If `!a.amount` is false, `a.amount === ''` will also always be false, so we only need one

**Review by jakubcolony — APPROVED** [review]
Nice work @joanna-pagepro, thanks for addressing all the feedback 👍

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Apologies @joanna-pagepro, I've just noticed this while testing #2039:

https://github.com/JoinColony/colonyCDapp/assets/112586815/21770a58-32f4-40fb-b6ce-8cc3d468d207

It seems to happen when you include a dot in the amount and then add a new row.

**Comment by joanna-pagepro** [comment]
@jakubcolony good you've noticed that! It should be ok now. Can you check again?

**Review by melyndav — APPROVED** [review]
Superstar effort @joanna-pagepro! Thank you for fixing all the updates, great work! 💯

**Review by jakubcolony — APPROVED** [review]
Nice work, the issue seems fixed! There's just a couple of comments outstanding from the previous reviews if you could have a look at those.

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx
Comment (joanna-pagepro) [reply src/components/v5/common/ActionSidebar/ActionSidebar.tsx:121]: If I remember it right, @CzarekDryl was working on it in a separate PR. The solution I added is temporary, only to make it possible to see the changes on mobile right now. I will remove it when Czarek's changes will be merged.

File: src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderPayoutsTotal/PaymentBuilderPayoutsTotal.tsx
Comment (joanna-pagepro) [reply src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/PaymentBuilderPayoutsTotal/PaymentBuilderPayoutsTotal.tsx]: Thanks for noticing, removed

**Pull request merged**
